provider/pkg/gen/schema.go

```diff
@@ -718,6 +723,19 @@ func (g *packageGenerator) genProperties(typeName string, typeSchema *discovery.
 	return &result, nil
 }
 
+func (g *packageGenerator) isCopyFromOutput(propName string) bool {
+	mod := strings.Split(g.mod, "/")[0]
+	switch mod {
+	case "compute", "container", "deploymentmanager":
+		return strings.Contains(strings.ToLower(propName), "fingerprint")
+	default:
+		// Some other modules contain properties called "fingerprint" or similar but that need to be
+		// populated manually by users, e.g. to provide a hash of a file or a certificate. Therefore,
+		// we maintain the opt-in list of modules above and treat the rest as normal inputs.
+		return false
```

Comment:
I am curious how we want to go about detecting that our opt-in/out lists are stale? Does it make sense to perhaps generate explicit lists in code and flag any deviation when we refresh discovery docs etc? Happy to track a ticket to track this if you like.

Reply:
Yes, that's a valid concern and a good idea to cut the ticket 👍

Reply:
Opened https://github.com/pulumi/pulumi-google-native/issues/133 to track this.